app/controllers/sellers_controller.rb

```diff
@@ -10,7 +10,11 @@ def show
   end
 
   def new
-    @seller = Seller.new
+    if session[:seller_id] != nil
+      redirect_to root_path 
+    else
+      @seller = Seller.new
+    end
```

Comment:
Also, if they're logged in, it might make sense to redirect them to their seller dashboard instead?